tensorflow/python/keras/layers/normalization.py

```diff
@@ -315,18 +352,22 @@ def build(self, input_shape):
         raise ValueError('When using virtual_batch_size, adjustment cannot '
                          'be specified')
 
+    fused_axis = self.axis
+    self._input_fused_shape = None
     if self.fused in (None, True):
-      # TODO(yaozhang): if input is not 4D, reshape it to 4D and reshape the
-      # output back to its original shape accordingly.
-      if self._USE_V2_BEHAVIOR:
-        if self.fused is None:
-          self.fused = (ndims == 4)
-        elif self.fused and ndims != 4:
-          raise ValueError('Batch normalization layers with fused=True only '
-                           'support 4D input tensors.')
-      else:
+      if len(self.axis) == 1 and (self.axis[0] not in (1, ndims - 1) or
+                                  ndims != 4):
+        # The fused implementation only supports NCHW or NHWC, so we will
+        # reshape the input/output tensor to/from an equivalent 4D shape.
+        fused_shape, fused_axis = self._get_shape_and_axis_for_fused(
+            input_shape.dims, self.axis[0])
+        fused_shape = tensor_shape.TensorShape(fused_shape)
```

Comment:
why convert this to a TensorShape then immediately convert back to a list?

Reply:
Done. (Twas a remnant of a fight I had initially with TensorShape vs. list, None vs. -1 etc.).